src/MCMCMonitorDataManager/updateSequences.ts

```diff
@@ -1,25 +1,39 @@
-import { GetSequencesRequest, isGetSequencesResponse } from "../../service/src/types";
+import { GetSequencesRequest, isGetSequencesResponse, MCMCSequenceUpdate } from "../../service/src/types";
 import postApiRequest from "../networking/postApiRequest";
+import getSpaSequenceUpdates from "../spaInterface/getSpaSequenceUpdates";
 import { MCMCMonitorAction, MCMCMonitorData } from "./MCMCMonitorDataTypes";
 
-
 export default async function updateSequences(data: MCMCMonitorData, dispatch: (a: MCMCMonitorAction) => void) {
     const X = data.sequences.filter(s => (s.updateRequested))
     if (X.length > 0) {
-        const req: GetSequencesRequest = {
-            type: 'getSequencesRequest',
-            sequences: X.map(s => ({
-                runId: s.runId, chainId: s.chainId, variableName: s.variableName, position: s.data.length
-            }))
+        const runId = X[0].runId
+        if (runId.startsWith('spa|')) {
+            // handle the special case of a stan playground run
+            const sequenceUpdates: MCMCSequenceUpdate[] | undefined = await getSpaSequenceUpdates(runId, X)
+            if (sequenceUpdates) {
+                dispatch({
+                    type: "updateSequenceData",
+                    sequences: sequenceUpdates
+                })
```

Comment:
```
const sequenceUpdtates = isSpaRunId(runId) ? ... : ...
```

Reply:
I am actually going to leave as an if branch for now, since the two branches are sufficiently different, but I will pull out the common dispatch call.